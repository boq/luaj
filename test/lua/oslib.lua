-- simple os-library tests
--
-- because the nature of the "os" library is to provide os-specific behavior,
-- the compatibility tests must be extremely loose, and can really only 
-- compare things like return value type to be meaningful.
-- 
-- actual os behavior needs to go in an oslib function test
-- 
local opcall = pcall
local pcall = function(...)
	local s,e,f = pcall(...)
	return s,type(e),type(f)
end 
print( 'os', type(os) )
print( 'os.clock()', pcall( os.clock ) )
print( 'os.date()', pcall( os.date ) )
print( 'os.difftime(123000, 21500)', pcall( os.difftime, 123000, 21250 ) )
print( 'os.getenv()', pcall( os.getenv ) )
print( 'os.getenv("bogus.key")', pcall( os.getenv, 'bogus.key' ) )
-- <broken by="design"> - pcall returns types instead of actual values
-- local s,p = pcall( os.tmpname )
-- local s,q = pcall( os.tmpname )
-- print( 'os.tmpname()', s, p )
-- print( 'os.tmpname()', s, q )
-- print( 'os.remove(p)', pcall( os.remove, p ) )
-- print( 'os.rename(p,q)', pcall( os.rename, p, q ) )
-- local s,f = pcall( io.open, p,"w" )
-- print( 'io.open', s, f )
-- print( 'write', pcall( f.write, f, "abcdef 12345" ) )
-- print( 'close', pcall( f.close, f ) )
-- print( 'os.rename(p,q)', pcall( os.rename, p, q ) )
-- print( 'os.remove(q)', pcall( os.remove, q ) )
-- print( 'os.remove(q)', pcall( os.remove, q ) )
-- </broken>
-- setlocale not supported on jse yet
-- print( 'os.setlocale()', pcall( os.setlocale ) )
-- print( 'os.setlocale("jp")', pcall( os.setlocale, "jp" ) )
-- print( 'os.setlocale("us","monetary")', pcall( os.setlocale, "us", "monetary" ) )
-- print( 'os.setlocale(nil,"all")', pcall( os.setlocale, nil, "all" ) )
print( 'os.setlocale("C")', pcall( os.setlocale, "C" ) )
print( 'os.exit', type(os.exit) )

-- os.date() formatting
local t = os.time{year=2010, month=8, day=9, hour=7, min=34, sec=56}
local function p(s)
  print( "os.date('"..s.."')", opcall(os.date, s, t))
end
for _,v in pairs{'%','a','A','b','B','c','d','H','I','j','m','M','p','S','U','w','W','x','X','y','Y'} do
  p('%'.. v)
end
local tbl = os.date('*t', t)
for i,k in ipairs({'year', 'month', 'day', 'hour', 'min', 'sec', 'wday', 'yday', 'isdst'}) do
	local v = tbl[k]
	print('k', type(k), k, 'v', type(v), v)
end

print('type(os.time())', type(os.time()))
print('os.date("%Y-%m-%d %H:%M:%S", os.time({year=1971, month=2, day=25}))', os.date("%Y-%m-%d %H:%M:%S", os.time({year=1971, month=2, day=25})))
print('os.date("%Y-%m-%d %H:%M:%S", os.time({year=1971, month=2, day=25, hour=11, min=22, sec=33}))', os.date("%Y-%m-%d %H:%M:%S", os.time({year=1971, month=2, day=25, hour=11, min=22, sec=33})))
