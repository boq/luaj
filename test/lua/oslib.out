os	table
os.clock()	true	number	nil
os.date()	true	string	nil
os.difftime(123000, 21500)	true	number	nil
os.getenv()	false	string	nil
os.getenv("bogus.key")	true	nil	nil
os.setlocale("C")	true	string	nil
os.exit	function
os.date('%%')	true	%
os.date('%a')	true	Mon
os.date('%A')	true	Monday
os.date('%b')	true	Aug
os.date('%B')	true	August
os.date('%c')	true	Mon Aug 09 07:34:56 2010
os.date('%d')	true	09
os.date('%H')	true	07
os.date('%I')	true	07
os.date('%j')	true	221
os.date('%m')	true	08
os.date('%M')	true	34
os.date('%p')	true	AM
os.date('%S')	true	56
os.date('%U')	true	32
os.date('%w')	true	1
os.date('%W')	true	32
os.date('%x')	true	08/09/10
os.date('%X')	true	07:34:56
os.date('%y')	true	10
os.date('%Y')	true	2010
k	string	year	v	number	2010
k	string	month	v	number	8
k	string	day	v	number	9
k	string	hour	v	number	7
k	string	min	v	number	34
k	string	sec	v	number	56
k	string	wday	v	number	2
k	string	yday	v	number	221
k	string	isdst	v	boolean	true
type(os.time())	number
os.date("%Y-%m-%d %H:%M:%S", os.time({year=1971, month=2, day=25}))	1971-02-25 12:00:00
os.date("%Y-%m-%d %H:%M:%S", os.time({year=1971, month=2, day=25, hour=11, min=22, sec=33}))	1971-02-25 11:22:33
